data/data/openstack/masters/private-network.tf

```diff
@@ -58,8 +61,13 @@ resource "openstack_networking_port_v2" "masters" {
     value = var.cluster_domain
   }
 
-  fixed_ip {
-    subnet_id = local.nodes_subnet_id
+  dynamic "fixed_ip" {
+    for_each = local.nodes_ports[count.index].fixed_ips
+
+    content {
+      subnet_id  = fixed_ip.value["subnet_id"]
+      ip_address = fixed_ip.value["ip_address"]
+    }
```

Comment:
This becomes a dynamic bloc, because with ports we are not limited any more to just one subnet per port. This array will have length!=1 if a failuredomain  defines a control-plane portTarget, AND that portTarget has multiple fixedIPs.